lib/inspec/file_provider.rb

```diff
@@ -105,6 +105,17 @@ def initialize(path)
       end
     end
 
+    def extract(destination_path = '.')
+      FileUtils.mkdir_p(destination_path)
+
+      Zip::File.open(@path) do |archive|
+        archive.each do |file|
+          final_path = File.join(destination_path, file.name)
+          archive.extract(file, final_path) unless File.exist?(final_path)
```

Comment:
Should we just overwrite the cache here incase its a newer version? Or maybe at least warn. Same with the next extract.

Reply:
I'm for clearing the directory and extracting cleanly.  If you had myprofile-0.1.0.zip, and you want to switch to myprofile-0.2.0.zip, you don't want the outcome to be "mostly 0.2.0, except we kept any files 0.1.0 had that 0.2.0 didn't".  That's bad, since we auto-load all library files and controls - anything there will be read, and comedy ensues.

Reply:
I think this would be okay because they are being extracted to a directory named after the checksum. That being said, it couldn't hurt to ensure the directory is fresh in case there is some manual tampering with the cache.

I'll make it so.